content/articles/2017-11-21-usethis-1-0-0.Rmd

```diff
@@ -0,0 +1,144 @@
+
+### Code and tests
+
+`use_r()` will create/edit a script file in `R/`:
+
+```{r}
+use_r("foo")
+```
+
+`use_test("foo")` will set up all the infrastructure you need for [unit testing](http://r-pkgs.had.co.nz/tests.html), and create/edit a test file in `tests/teststhat/`:
+
+```{r}
+use_test("foo")
+```
+
+We recommend that you have paired test and source files. If you use RStudio, you can use `use_test()` to create/edit the test file corresponding to the open source file, and `use_r()` to create/edit to the source file corresponding to the open test file. 
+
+### Dependencies
+
+To declare that your project has some dependencies, use `use_package()`:
+
+```{r}
+use_package("ggplot2")
+use_package("dplyr")
+```
+
+Or if you want a dependency on a development version of a package, use `use_dev_package()`.
+
+### Documentation
+
+Take advantage of these helpers to document your package:
+
+* `use_roxygen_md()` sets up [roxygen2](http://r-pkgs.had.co.nz/man.html) and
+  enables markdown mode so you can use markdown in your roxygen2 comment blocks.
+
+* `use_package_doc()` creates a skeleton documentation file for the complete
+  package, taking the advantage of the latest roxygen2 features to minimise
+  duplication between the `DESCRIPTION` and the documentation.
+  
+* `use_readme_rmd()` creates a
+  [`README.Rmd`](http://r-pkgs.had.co.nz/release.html#important-files): 
+  use this to describe what your package does and why people should care about 
+  it.
+  
+* `use_news_md()` creates a basic `NEWS.md` for you to record changes.
+
+* `use_vignette("vignette-name")` sets you up for success by configuring
+  `DESCRIPTION` and creating a `.Rmd` template in `vignettes/`
+
+### Sharing 
+
+If you want to share your code with others, it's good practice to make the licensing clear. usethis provides four helpers for four most common open source licenses:
+
+* [`use_mit_license()`](https://choosealicense.com/licenses/mit)
+* [`use_apl2_license()`](https://choosealicense.com/licenses/apache-2.0/)
+* [`use_gpl3_license()`](https://choosealicense.com/licenses/gpl-3.0/)
+* [`use_cc0_license()`](https://creativecommons.org/publicdomain/zero/1.0/)
+
+These set the `License` field in the `DESCRIPTION` and include the license text in  `LICENSE.md`, which is then added to `.Rbuildignore`. This allows you to follow licensing best practices while adhering to CRAN's requirements.
+
+For this package, I'll use an MIT license which is simple and permissive:
+
+```{r}
+use_mit_license("Hadley Wickham")
+```
+
+It's also easy to activate git:
+
+```{r}
+use_git()
+```
+
+And publish to GitHub (I'm not going to run this code here!)
+
+```{r, eval = FALSE}
+use_github()
+```
+
+(For this to work, you'll need to set a `GITHUB_PAT` environment variable in your `~/.Renviron`. Follow [Jenny Bryan's instructions](https://github.com/r-lib/gh/blob/master/R/gh_whoami.R#L7-L25), and use `edit_r_environ()` to easily access the right file for editing)
+
```

Comment:
I didn't realize why you were asking me this. Here's another version that is easier on the eyes:

http://happygitwithr.com/github-pat.html#step-by-step

Reply:
Much better, thanks!